src/FreeAge/client/render_window.cpp

```diff
@@ -3577,6 +3718,11 @@ void RenderWindow::paintGL() {
         continue;
       }
       
+      // Drop if the object have been garrisoned.
```

Comment:
In that case the object would be selected (at least what happens in the original)

Reply:
You are right, the idle-villager function might then need an internal counter to keep track of which the current/next villager to select is (or just treat all idle villagers within the same object as a single group), but the selection should be on the object.

(This may cause some unintended consequences: When deleting some villagers late-game to make space for more military, and quickly hitting the idle-villager button and the delete button alternatingly to accomplish that, then if a villager happens to be garrisoned in a castle, some accident may happen. But that's unrelated to the PR here and may also be addressed by making a deletion warning.)